src/NaadRepositoryClient.php

```
@@ -31,54 +39,65 @@ public function __construct()
     }
 
     /**
-     * Per NAAD documentation, certain characters must be replaced when
-     * building the URL for the short-term alert repository.
+     * Fetches an alert from the NAAD alert repository.
      *
-     * @param string $s The string to perform the replacement on.
+     * @param array $reference Heartbeat references array parts (sender, id, sent).
      *
-     * @return string
+     * @return string|null The alert response body or null if an error occurs.
+     *
+     * @throws Exception if an error occurs during the GET request.
      */
-    protected function replaceUrlCharacters( string $s ): string
+    public function fetchAlert(array $reference): ?string
     {
-        return str_replace(
-            [ '-', '+', ':' ],
-            [ '_', 'p', '_' ],
-            $s
-        );
+        $url = $this->getURL($reference);
+        $this->client = new Client();
+
+        try {
+            $response = $this->client->get($url);
+
+            return (string) $response->getBody();
+        } catch (RequestException $e) {
+            error_log("Request failed: " . $e->getMessage());
+            return null;
```

Can we have this throw the Exception instead of returning null so we don't have to change the SocketClient::validateAlert() arguments?